File: AOC-2021/Day1/Day1.js
// Day 1: Sonar Sweep
// https://adventofcode.com/2021/day/1

const fs = require("fs");
const file = fs.readFileSync("AOC-2021/Day1/input.txt");
const inputArray = file.toString().split("\n").map((val) => parseInt(val));

const testArray = [199, 200, 208, 210, 200, 207, 240, 269, 260, 263];

function solvePart1(arr) {
  let count = 0;
  for(let i = 1; i < arr.length; i++) {
    if (arr[i] > arr[i - 1]) count++;
  }
  return count;
}

console.log("Part 1: " + solvePart1(inputArray));

function solvePart2(arr) {
  let prevSum = 0;
  for(let i = 0; i < 3; i++) {
    prevSum += arr[i];
  }

  let count = 0;
  for(let i = 3; i < arr.length; i++) {
    const newSum = prevSum + arr[i] - arr[i - 3];
    if (newSum > prevSum) count++;
    prevSum = newSum;
  }
  return count;
}

console.log("Part 2: " + solvePart2(inputArray));

File: AOC-2022/Day1/Day1.py
# Day 1: Calorie Counting
# https://adventofcode.com/2022/day/1

with open("AOC-2022/Day1/input.txt") as f:
  raw_input = f.read()
  carries = [sum(int(x) for x in line.strip().split("\n")) for line in raw_input.split("\n\n")]

  # Part 1
  print("Part 1:", max(carries))

  # Part 2
  print("Part 2:", sum(sorted(carries)[-3:]))
